### ISSUE TITLE
ServiceFabricComposeDeploy does not support Service Fabric SDK 2.8

### ISSUE THREAD

**Issue by MikkelHegn** [Microsoft/vsts-tasks#5505]
The current ServiceFabricComposeDeploy tasks does not support Service Fabric SDK 2.8, which targets Service Fabric runtime 6.0. You can still use the 2.7 SDK to deploy to a 6.0 runtime cluster, however there should be a new version of the task , which supports the 2.8 SDK, as the commands have changed.

https://github.com/Azure/service-fabric-issues/issues/491

**Comment by brandmo** [comment]
Hi there, closed item #491 suggests this problem should be resolved now, but I tried again with SDK 2.8 on my build server and got the following error "##[error]The term 'Get-ServiceFabricComposeApplicationStatus' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again."
Will a new version of ServiceFabricComposeDeploy task be released soon?

**Comment by NCarlsonMSFT** [comment]
Version 0.2.1 of the task should be using Get-ServiceFabricComposeDeploymentStatus on a machine with the 2.8 SDK installed. Could you run your build with System.Debug set to true and include the output?

**Comment by brandmo** [comment]
Looks like it's using version 0.1.7 of the task and not 0.2.1. Is there anything I can do to make it use the correct version?

**Comment by mdegroot71** [comment]
Here's build output with System.Debug set to true.

2017-11-01T20:42:36.6464339Z ##[section]Starting: Release
2017-11-01T20:42:36.6484337Z Current agent version: '2.123.0'
2017-11-01T20:42:36.6564348Z ##[debug]Cleaning agent temp folder: d:\a\_temp
2017-11-01T20:42:36.6954595Z ##[debug]SET TMP=d:\a\_temp
2017-11-01T20:42:36.6954595Z ##[debug]SET TEMP=d:\a\_temp
2017-11-01T20:42:37.2948670Z ##[section]Starting: Initialize Job
2017-11-01T20:42:37.2968662Z Prepare release directory.
2017-11-01T20:42:37.2978667Z ReleaseId=14, TeamProjectId=ef9a454a-b57b-47a1-af23-2652f95ed480, ReleaseDefinitionName=fabrikamfiber - 3
2017-11-01T20:42:37.3088669Z Release folder: d:\a\r1\a
2017-11-01T20:42:37.3238666Z Environment variables available are below.  Note that these environment variables can be referred to in the task (in the ReleaseDefinition) by replacing "_" with "." e.g. AGENT_NAME environment variable can be referenced using Agent.Name in the ReleaseDefinition: 
				[AGENT_HOMEDIRECTORY] --> [C:\LR\mms\Services\Mms\Provisioner\TaskAgent\agents\2.123.0]
				[AGENT_ID] --> [3]
				[AGENT_JOBNAME] --> [Release]
				[AGENT_MACHINENAME] --> [factoryvm-az204]
				[AGENT_NAME] --> [Hosted Agent]
				[AGENT_OS] --> [Windows_NT]
				[AGENT_RELEASEDIRECTORY] --> [d:\a\r1\a]
				[AGENT_ROOTDIRECTORY] --> [d:\a]
				[AGENT_SERVEROMDIRECTORY] --> [C:\LR\mms\Services\Mms\Provisioner\TaskAgent\agents\2.123.0\externals\vstsom]
				[AGENT_TEMPDIRECTORY] --> [d:\a\_temp]
				[AGENT_TOOLSDIRECTORY] --> [d:\a\_tool]
				[AGENT_VERSION] --> [2.123.0]
				[AGENT_WORKFOLDER] --> [d:\a]
				[AZURE_HTTP_USER_AGENT] --> [VSTS_434bf39b-ee13-47d0-b0d1-1b53a27ad73f_release_5_14_14_1]
				[BUILD_BUILDID] --> [10]
				[BUILD_BUILDNUMBER] --> [20171101.1]
				[BUILD_BUILDURI] --> [vstfs:///Build/Build/10]
				[BUILD_DEFINITIONID] --> [5]
				[BUILD_DEFINITIONNAME] --> [fabrikamfiber - 3]
				[BUILD_PROJECTID] --> [ef9a454a-b57b-47a1-af23-2652f95ed480]
				[BUILD_PROJECTNAME] --> [FabrikamFiber]
				[BUILD_REPOSITORY_ID] --> [78c38456-021b-4cca-a146-ac5091f22f4b]
				[BUILD_REPOSITORY_NAME] --> [FabrikamFiber]
				[BUILD_REPOSITORY_PROVIDER] --> [TfsGit]
				[BUILD_REQUESTEDFOR] --> [Michael de Groot]
				[BUILD_REQUESTEDFORID] --> [46754d72-0326-62e5-8482-c1c9c464d4ca]
				[BUILD_SOURCEBRANCH] --> [refs/heads/master]
				[BUILD_SOURCEBRANCHNAME] --> [master]
				[BUILD_SOURCEVERSION] --> [bf863cd932696c26bbf624b226eaa195ea1951e5]
				[BUILD_TYPE] --> [Build]
				[MSDEPLOY_HTTP_USER_AGENT] --> [VSTS_434bf39b-ee13-47d0-b0d1-1b53a27ad73f_release_5_14_14_1]
				[RELEASE_ARTIFACTS_DROP_BUILDID] --> [10]
				[RELEASE_ARTIFACTS_DROP_BUILDNUMBER] --> [20171101.1]
				[RELEASE_ARTIFACTS_DROP_BUILDURI] --> [vstfs:///Build/Build/10]
				[RELEASE_ARTIFACTS_DROP_DEFINITIONID] --> [5]
				[RELEASE_ARTIFACTS_DROP_DEFINITIONNAME] --> [fabrikamfiber - 3]
				[RELEASE_ARTIFACTS_DROP_PROJECTID] --> [ef9a454a-b57b-47a1-af23-2652f95ed480]
				[RELEASE_ARTIFACTS_DROP_PROJECTNAME] --> [FabrikamFiber]
				[RELEASE_ARTIFACTS_DROP_REPOSITORY_ID] --> [78c38456-021b-4cca-a146-ac5091f22f4b]
				[RELEASE_ARTIFACTS_DROP_REPOSITORY_NAME] --> [FabrikamFiber]
				[RELEASE_ARTIFACTS_DROP_REPOSITORY_PROVIDER] --> [TfsGit]
				[RELEASE_ARTIFACTS_DROP_REQUESTEDFOR] --> [Michael de Groot]
				[RELEASE_ARTIFACTS_DROP_REQUESTEDFORID] --> [46754d72-0326-62e5-8482-c1c9c464d4ca]
				[RELEASE_ARTIFACTS_DROP_SOURCEBRANCH] --> [refs/heads/master]
				[RELEASE_ARTIFACTS_DROP_SOURCEBRANCHNAME] --> [master]
				[RELEASE_ARTIFACTS_DROP_SOURCEVERSION] --> [bf863cd932696c26bbf624b226eaa195ea1951e5]
				[RELEASE_ARTIFACTS_DROP_TYPE] --> [Build]
				[RELEASE_ATTEMPTNUMBER] --> [1]
				[RELEASE_DEFINITIONENVIRONMENTID] --> [5]
				[RELEASE_DEFINITIONID] --> [5]
				[RELEASE_DEFINITIONNAME] --> [fabrikamfiber - 3]
				[RELEASE_DEPLOYMENT_REQUESTEDFOR] --> [Michael de Groot]
				[RELEASE_DEPLOYMENT_REQUESTEDFOREMAIL] --> [MDeGroot@10thmagnitude.com]
				[RELEASE_DEPLOYMENT_REQUESTEDFORID] --> [46754d72-0326-62e5-8482-c1c9c464d4ca]
				[RELEASE_DEPLOYMENTID] --> [16]
				[RELEASE_DEPLOYPHASEID] --> [16]
				[RELEASE_ENVIRONMENTID] --> [14]
				[RELEASE_ENVIRONMENTNAME] --> [Dev]
				[RELEASE_ENVIRONMENTS_DEV_STATUS] --> [InProgress]
				[RELEASE_ENVIRONMENTURI] --> [vstfs:///ReleaseManagement/Environment/14]
				[RELEASE_REASON] --> [ContinuousIntegration]
				[RELEASE_RELEASEDESCRIPTION] --> [Triggered by fabrikamfiber - 3 20171101.1.]
				[RELEASE_RELEASEID] --> [14]
				[RELEASE_RELEASENAME] --> [Release-1]
				[RELEASE_RELEASEURI] --> [vstfs:///ReleaseManagement/Release/14]
				[RELEASE_RELEASEWEBURL] --> [https://mdegroot-10thmagnitude.visualstudio.com/ef9a454a-b57b-47a1-af23-2652f95ed480/_release?releaseId=14&_a=release-summary]
				[RELEASE_REQUESTEDFOR] --> [Michael de Groot]
				[RELEASE_REQUESTEDFOREMAIL] --> [MDeGroot@10thmagnitude.com]
				[RELEASE_REQUESTEDFORID] --> [46754d72-0326-62e5-8482-c1c9c464d4ca]
				[RELEASE_SKIPARTIFACTSDOWNLOAD] --> [False]
				[REQUESTEDFORID] --> [46754d72-0326-62e5-8482-c1c9c464d4ca]
				[SYSTEM] --> [release]
				[SYSTEM_ARTIFACTSDIRECTORY] --> [d:\a\r1\a]
				[SYSTEM_COLLECTIONID] --> [434bf39b-ee13-47d0-b0d1-1b53a27ad73f]
				[SYSTEM_CULTURE] --> [en-US]
				[SYSTEM_DEBUG] --> [true]
				[SYSTEM_DEFAULTWORKINGDIRECTORY] --> [d:\a\r1\a]
				[SYSTEM_ENABLEACCESSTOKEN] --> [False]
				[SYSTEM_HOSTTYPE] --> [release]
				[SYSTEM_JOBID] --> [5043bddc-0142-4ccd-9bd4-33906a44ff4b]
				[SYSTEM_JOBPOSITIONINPHASE] --> [1]
				[SYSTEM_PARALLELEXECUTIONTYPE] --> [None]
				[SYSTEM_PLANID] --> [68804a33-3766-4402-9a7d-33182f08d53f]
				[SYSTEM_TASKDEFINITIONSURI] --> [https://mdegroot-10thmagnitude.visualstudio.com/]
				[SYSTEM_TEAMFOUNDATIONCOLLECTIONURI] --> [https://mdegroot-10thmagnitude.visualstudio.com/]
				[SYSTEM_TEAMFOUNDATIONSERVERURI] --> [https://mdegroot-10thmagnitude.vsrm.visualstudio.com/]
				[SYSTEM_TEAMPROJECT] --> [FabrikamFiber]
				[SYSTEM_TEAMPROJECTID] --> [ef9a454a-b57b-47a1-af23-2652f95ed480]
				[SYSTEM_TIMELINEID] --> [68804a33-3766-4402-9a7d-33182f08d53f]
				[SYSTEM_TOTALJOBSINPHASE] --> [1]
				[SYSTEM_WORKFOLDER] --> [d:\a]
2017-11-01T20:42:37.3278661Z Download all required tasks.
2017-11-01T20:42:37.3418662Z Downloading task: ServiceFabricComposeDeploy
2017-11-01T20:42:37.6064410Z ##[debug]Task 'ServiceFabricComposeDeploy' has been downloaded into 'd:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7'.
2017-11-01T20:42:38.4006038Z ##[debug]Task 'Deploy docker-compose application to a Service Fabric cluster' has following condition: 'succeeded()'.
2017-11-01T20:42:38.4065741Z ##[debug]Parsing expression: <succeeded()>
2017-11-01T20:42:38.4116083Z ##[debug]succeeded
2017-11-01T20:42:38.4126091Z ##[debug](
2017-11-01T20:42:38.4136001Z ##[debug])
2017-11-01T20:42:38.4325990Z ##[section]Finishing: Initialize Job
2017-11-01T20:42:38.4495780Z ##[debug]Evaluating condition for step: 'Download Artifacts'
2017-11-01T20:42:38.4525738Z ##[debug]Evaluating: succeeded()
2017-11-01T20:42:38.4535736Z ##[debug]Evaluating succeeded:
2017-11-01T20:42:38.4565736Z ##[debug]=> (Boolean) True
2017-11-01T20:42:38.4595737Z ##[debug]Expanded: True
2017-11-01T20:42:38.4595737Z ##[debug]Result: True
2017-11-01T20:42:38.4655733Z ##[section]Starting: Download Artifacts
2017-11-01T20:42:41.2606362Z Downloading artifact
2017-11-01T20:42:41.2666356Z Creating artifacts directory: d:\a\r1\a
2017-11-01T20:42:41.2686357Z Created artifacts directory: d:\a\r1\a
2017-11-01T20:42:41.2706366Z Starting artifacts download...
2017-11-01T20:42:41.2736354Z Downloading linked artifact Drop of type Build...
2017-11-01T20:42:41.2770721Z Ensuring artifact folder d:\a\r1\a\Drop exists and is clean.
2017-11-01T20:42:41.3024291Z Preparing to get the list of available artifacts from build
2017-11-01T20:42:41.7591366Z Preparing to download artifact: docker-compose
2017-11-01T20:42:41.7741660Z Artifact Type: ServerDrop
2017-11-01T20:42:41.7751373Z Downloading artifact from file container: #/324708/docker-compose to target: d:\a\r1\a\Drop\docker-compose
2017-11-01T20:42:41.7751373Z Parallel download limit: 4
2017-11-01T20:42:41.7751373Z Download buffer size: 8192
2017-11-01T20:42:42.0635234Z Caching items under 'docker-compose' in the file container...
2017-11-01T20:42:42.1473527Z Caching complete. (83 ms)
2017-11-01T20:42:42.1543520Z ##[debug][File] docker-compose/docker-compose.yml => d:\a\r1\a\Drop\docker-compose\docker-compose.yml
2017-11-01T20:42:42.1573554Z ##[debug]Acquiring semaphore to download file d:\a\r1\a\Drop\docker-compose\docker-compose.yml
2017-11-01T20:42:42.1583563Z Downloading file d:\a\r1\a\Drop\docker-compose\docker-compose.yml
2017-11-01T20:42:42.1653546Z ##[debug]Get file container client for file docker-compose/docker-compose.yml
2017-11-01T20:42:42.1653546Z ##[debug]Start fetch file stream from filecontainer service for file docker-compose/docker-compose.yml
2017-11-01T20:42:42.1763527Z ##[debug]Fetching contents of file d:\a\r1\a\Drop\docker-compose\docker-compose.yml.download
2017-11-01T20:42:42.2612573Z ##[debug]Finished fetch file stream from filecontainer service for file docker-compose/docker-compose.yml
2017-11-01T20:42:42.2612573Z ##[debug]Writing contents of file d:\a\r1\a\Drop\docker-compose\docker-compose.yml.download to disk
2017-11-01T20:42:42.2652303Z ##[debug]Finished writing contents of file d:\a\r1\a\Drop\docker-compose\docker-compose.yml.download to disk
2017-11-01T20:42:42.2672425Z Download complete.
2017-11-01T20:42:42.2682577Z 1 placed file(s): 1 downloaded, 0 empty
2017-11-01T20:42:42.2722298Z 1 MB downloaded at 2 KB/sec. Download time: 00:00:00.1125122.
2017-11-01T20:42:42.2722298Z Downloaded linked artifact Drop
2017-11-01T20:42:42.2722298Z Finished artifacts download
2017-11-01T20:42:42.2722298Z Downloading commits
2017-11-01T20:42:42.2762305Z ##[section]Finishing: Download Artifacts
2017-11-01T20:42:42.2902511Z ##[debug]Evaluating condition for step: 'Deploy docker-compose application to a Service Fabric cluster'
2017-11-01T20:42:42.2912305Z ##[debug]Evaluating: succeeded()
2017-11-01T20:42:42.2912305Z ##[debug]Evaluating succeeded:
2017-11-01T20:42:42.2912305Z ##[debug]=> (Boolean) True
2017-11-01T20:42:42.2912305Z ##[debug]Expanded: True
2017-11-01T20:42:42.2912305Z ##[debug]Result: True
2017-11-01T20:42:42.2912305Z ##[section]Starting: Deploy docker-compose application to a Service Fabric cluster
2017-11-01T20:42:42.3092304Z ==============================================================================
2017-11-01T20:42:42.3102297Z Task         : Service Fabric Compose Deploy
2017-11-01T20:42:42.3102297Z Description  : Deploy a docker-compose application to a Service Fabric cluster.
2017-11-01T20:42:42.3102297Z Version      : 0.1.7
2017-11-01T20:42:42.3102297Z Author       : Microsoft Corporation
2017-11-01T20:42:42.3102297Z Help         : [More Information](https://go.microsoft.com/fwlink/?LinkID=847030)
2017-11-01T20:42:42.3102297Z ==============================================================================
2017-11-01T20:42:43.7698034Z ##[debug]VstsTaskSdk 0.8.2 commit 
2017-11-01T20:42:44.1184721Z ##[debug]Entering D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ServiceFabricComposeDeploy.ps1.
2017-11-01T20:42:44.1290095Z ##[debug]Loading resource strings from: D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\task.json
2017-11-01T20:42:44.1938551Z ##[debug]Loaded 15 strings.
2017-11-01T20:42:44.1998537Z ##[debug]SYSTEM_CULTURE: 'en-US'
2017-11-01T20:42:44.2008535Z ##[debug]Loading resource strings from: D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\Strings\resources.resjson\en-US\resources.resjson
2017-11-01T20:42:44.2378544Z ##[debug]Loaded 15 strings.
2017-11-01T20:42:44.2428572Z ##[debug]Loading module from path 'D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ps_modules\ServiceFabricHelpers\ServiceFabricHelpers.psm1'.
2017-11-01T20:42:44.2478546Z ##[debug]Loading resource strings from: D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ps_modules\ServiceFabricHelpers\module.json
2017-11-01T20:42:44.2738533Z ##[debug]Loaded 11 strings.
2017-11-01T20:42:44.2768526Z ##[debug]SYSTEM_CULTURE: 'en-US'
2017-11-01T20:42:44.2798542Z ##[debug]Loading resource strings from: D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ps_modules\ServiceFabricHelpers\Strings\resources.resjson\en-US\resources.resjson
2017-11-01T20:42:44.3708547Z ##[debug]Loaded 11 strings.
2017-11-01T20:42:44.3918533Z ##[debug]Entering Find-VstsFiles.
2017-11-01T20:42:44.4038550Z ##[debug] LiteralDirectory: 'D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ps_modules\ServiceFabricHelpers'
2017-11-01T20:42:44.4058529Z ##[debug] LegacyPattern: '*.ps1'
2017-11-01T20:42:44.8657254Z ##[debug]Entering Get-MatchingItems.
2017-11-01T20:42:44.8697265Z ##[debug] IncludePatterns: 'D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ps_modules\ServiceFabricHelpers\*.ps1'
2017-11-01T20:42:44.8757290Z ##[debug] ExcludePatterns: ''
2017-11-01T20:42:44.8797257Z ##[debug] IncludeFiles: 'True'
2017-11-01T20:42:44.8817265Z ##[debug] IncludeDirectories: 'False'
2017-11-01T20:42:44.8837264Z ##[debug] Force: 'False'
2017-11-01T20:42:45.3523124Z ##[debug]Paths: D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ps_modules\ServiceFabricHelpers\
2017-11-01T20:42:45.3933128Z ##[debug] Connect-ServiceFabricClusterFromServiceEndpoint.ps1
2017-11-01T20:42:45.3933128Z ##[debug] Get-ServiceFabricEncryptedText.ps1
2017-11-01T20:42:45.3933128Z ##[debug]Leaving Get-MatchingItems.
2017-11-01T20:42:45.3933128Z ##[debug]Total found: 2
2017-11-01T20:42:45.3933128Z ##[debug]Leaving Find-VstsFiles.
2017-11-01T20:42:45.3943127Z ##[debug]Exporting function 'Connect-ServiceFabricClusterFromServiceEndpoint'.
2017-11-01T20:42:45.3943127Z ##[debug]Exporting function 'Get-ServiceFabricEncryptedText'.
2017-11-01T20:42:45.3943127Z ##[debug]Importing function 'Connect-ServiceFabricClusterFromServiceEndpoint'.
2017-11-01T20:42:45.3943127Z ##[debug]Importing function 'Get-ServiceFabricEncryptedText'.
2017-11-01T20:42:45.4393126Z ##[debug]INPUT_SERVICECONNECTIONNAME: 'bb44fe9e-7ca5-48bc-83c3-72acad648d37'
2017-11-01T20:42:45.4403122Z ##[debug]ENDPOINT_URL_bb44fe9e-7ca5-48bc-83c3-72acad648d37: 'tcp://mdgsf2.eastus.cloudapp.azure.com:19000/'
2017-11-01T20:42:45.4413124Z ##[debug]ENDPOINT_AUTH_bb44fe9e-7ca5-48bc-83c3-72acad648d37: '********'
2017-11-01T20:42:45.4413124Z ##[debug]ENDPOINT_DATA_bb44fe9e-7ca5-48bc-83c3-72acad648d37: '{}'
2017-11-01T20:42:45.4413124Z ##[debug]INPUT_COMPOSEFILEPATH: 'd:\a\r1\a\**\docker-compose.yml'
2017-11-01T20:42:45.4413124Z Searching for path: d:\a\r1\a\**\docker-compose.yml
2017-11-01T20:42:45.4423119Z ##[debug]Entering Find-VstsFiles.
2017-11-01T20:42:45.4423119Z ##[debug] LegacyPattern: 'd:\a\r1\a\**\docker-compose.yml'
2017-11-01T20:42:45.4423119Z ##[debug]Entering Get-MatchingItems.
2017-11-01T20:42:45.4423119Z ##[debug] IncludePatterns: 'd:\a\r1\a\**\docker-compose.yml'
2017-11-01T20:42:45.4423119Z ##[debug] ExcludePatterns: ''
2017-11-01T20:42:45.4423119Z ##[debug] IncludeFiles: 'True'
2017-11-01T20:42:45.4863119Z ##[debug] IncludeDirectories: 'False'
2017-11-01T20:42:45.6153114Z ##[debug] Force: 'False'
2017-11-01T20:42:45.6303118Z ##[debug]Path: d:\a\r1\a\Drop\docker-compose\docker-compose.yml
2017-11-01T20:42:45.6303118Z ##[debug]Leaving Get-MatchingItems.
2017-11-01T20:42:45.6303118Z ##[debug]Total found: 1
2017-11-01T20:42:45.6303118Z ##[debug]Leaving Find-VstsFiles.
2017-11-01T20:42:45.6303118Z ##[debug]Asserting leaf path exists: 'd:\a\r1\a\Drop\docker-compose\docker-compose.yml'
2017-11-01T20:42:45.6303118Z Found path: d:\a\r1\a\Drop\docker-compose\docker-compose.yml
2017-11-01T20:42:45.6313115Z ##[debug]INPUT_APPLICATIONNAME: 'fabric:/fabrikamfiber'
2017-11-01T20:42:45.6313115Z ##[debug]INPUT_DEPLOYTIMEOUTSEC (empty)
2017-11-01T20:42:45.6313115Z ##[debug]INPUT_REMOVETIMEOUTSEC (empty)
2017-11-01T20:42:45.6313115Z ##[debug]INPUT_GETSTATUSTIMEOUTSEC (empty)
2017-11-01T20:42:45.6313115Z Checking compose file
2017-11-01T20:42:47.9307667Z ##[debug]Join-Path "C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code" "ServiceFabricServiceModel.xsd"
2017-11-01T20:42:47.9357739Z ##[debug]C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code\ServiceFabricServiceModel.xsd
2017-11-01T20:42:47.9377961Z ##[debug]Length: 94
2017-11-01T20:42:47.9387978Z ##[debug]
2017-11-01T20:42:47.9407807Z ##[debug]Test-Path "C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code\ServiceFabricServiceModel.xsd"
2017-11-01T20:42:47.9437774Z ##[debug]True
2017-11-01T20:42:47.9447726Z ##[debug]
2017-11-01T20:42:48.3940807Z ##[warning]The Docker compose file contains the following 'keys' which are not supported. They will be ignored. 
'build' 
2017-11-01T20:42:48.3940807Z ##[debug]Processed: ##vso[task.logissue type=warning]The Docker compose file contains the following 'keys' which are not supported. They will be ignored. %0D%0A'build' 
2017-11-01T20:42:48.4010795Z ##[debug]Entering Connect-ServiceFabricClusterFromServiceEndpoint.
2017-11-01T20:42:48.4091065Z ##[debug] ClusterConnectionParameters: 'System.Collections.Hashtable'
2017-11-01T20:42:48.4130810Z ##[debug] ConnectedServiceEndpoint: '@{Url=tcp://mdgsf2.eastus.cloudapp.azure.com:19000/; Data=; Auth=}'
2017-11-01T20:42:48.6296371Z Imported cluster client certificate with thumbprint '********'.
2017-11-01T20:42:48.6416370Z 
2017-11-01T20:42:48.6422821Z Thumbprint                                Subject                                                                      
2017-11-01T20:42:48.6422821Z ----------                                -------                                                                      
2017-11-01T20:42:48.6422821Z ********  CN=mdgsfcluster.EastUS.cloudapp.azure.com                                    
2017-11-01T20:42:51.1286669Z Successfully connected to cluster.
2017-11-01T20:42:51.1346708Z ##[debug]Leaving Connect-ServiceFabricClusterFromServiceEndpoint.
2017-11-01T20:42:51.1366710Z ##[debug]INPUT_REGISTRYCREDENTIALS: 'AzureResourceManagerEndpoint'
2017-11-01T20:42:51.1406707Z ##[debug]INPUT_AZURESUBSCRIPTIONENDPOINT: '9d4ea01f-eda8-4c01-8476-97ff8d704c55'
2017-11-01T20:42:51.1426713Z ##[debug]ENDPOINT_URL_9d4ea01f-eda8-4c01-8476-97ff8d704c55: 'https://management.azure.com/'
2017-11-01T20:42:51.1446743Z ##[debug]ENDPOINT_AUTH_9d4ea01f-eda8-4c01-8476-97ff8d704c55: '********'
2017-11-01T20:42:51.1466742Z ##[debug]ENDPOINT_DATA_9d4ea01f-eda8-4c01-8476-97ff8d704c55: '{"subscriptionId":"16a4217e-fd95-4b56-9dc6-a6fb8c1dce71","subscriptionName":"Free Trial","creationMode":"Automatic","environment":"AzureCloud","appObjectId":"deb51619-f3b3-42eb-9027-7dbb5b9c93cd","spnObjectId":"76efd90d-514f-4fb3-a191-9543685e3a49","azureSpnPermissions":"[{\"roleAssignmentId\":\"27caf654-a318-499e-bf63-802154674778\",\"resourceProvider\":\"Microsoft.RoleAssignment\",\"provisioned\":true}]","azureSpnRoleAssignmentId":"27caf654-a318-499e-bf63-802154674778","environmentUrl":"https://management.azure.com/","galleryUrl":"https://gallery.azure.com/","serviceManagementUrl":"https://management.core.windows.net/","resourceManagerUrl":"https://management.azure.com/","activeDirectoryAuthority":"https://login.microsoftonline.com/","environmentAuthorityUrl":"https://login.windows.net/","graphUrl":"https://graph.windows.net/","managementPortalUrl":"https://manage.windowsazure.com/","armManagementPortalUrl":"https://portal.azure.com/","activeDirectoryServiceEndpointResourceId":"https://management.core.windows.net/","sqlDatabaseDnsSuffix":".database.windows.net","AzureKeyVaultDnsSuffix":"vault.azure.net","AzureKeyVaultServiceEndpointResourceId":"https://vault.azure.net","StorageEndpointSuffix":"core.windows.net"}'
2017-11-01T20:42:51.1509425Z Encrypting the password with the Server Certificate.
2017-11-01T20:42:52.5644576Z ##[debug]Leaving D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ServiceFabricComposeDeploy.ps1.
2017-11-01T20:42:52.6363072Z ##[debug]Caught exception from task script.
2017-11-01T20:42:52.6393326Z ##[debug]Error record:
2017-11-01T20:42:52.6943440Z ##[debug]Get-ServiceFabricComposeApplicationStatusHelper : The term 'Get-ServiceFabricComposeApplicationStatus' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2017-11-01T20:42:52.6953185Z ##[debug]At D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ServiceFabricComposeDeploy.ps1:123 char:28
2017-11-01T20:42:52.6963069Z ##[debug]+ ... plication = Get-ServiceFabricComposeApplicationStatusHelper -UsePrevi ...
2017-11-01T20:42:52.6983166Z ##[debug]+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
2017-11-01T20:42:52.6993194Z ##[debug]    + CategoryInfo          : ObjectNotFound: (Get-ServiceFabr...plicationStatus:String) [Get-ServiceFabr...ionStatusH    elper], CommandNotFoundException
2017-11-01T20:42:52.7003162Z ##[debug]    + FullyQualifiedErrorId : CommandNotFoundException,Get-ServiceFabricComposeApplicationStatusHelper
2017-11-01T20:42:52.7013077Z ##[debug] 
2017-11-01T20:42:52.7023448Z ##[debug]Script stack trace:
2017-11-01T20:42:52.7063188Z ##[debug]at Get-ServiceFabricComposeApplicationStatusHelper, D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\utilities.ps1: line 74
2017-11-01T20:42:52.7073339Z ##[debug]at <ScriptBlock>, D:\a\_tasks\ServiceFabricComposeDeploy_19c02b15-d377-40e0-9efa-3168506e0933\0.1.7\ServiceFabricComposeDeploy.ps1: line 123
2017-11-01T20:42:52.7083172Z ##[debug]at <ScriptBlock>, <No file>: line 1
2017-11-01T20:42:52.7093428Z ##[debug]at <ScriptBlock>, <No file>: line 22
2017-11-01T20:42:52.7103068Z ##[debug]at <ScriptBlock>, <No file>: line 18
2017-11-01T20:42:52.7113185Z ##[debug]at <ScriptBlock>, <No file>: line 1
2017-11-01T20:42:52.7133435Z ##[debug]Exception:
2017-11-01T20:42:52.7173328Z ##[debug]System.Management.Automation.CommandNotFoundException: The term 'Get-ServiceFabricComposeApplicationStatus' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2017-11-01T20:42:52.7183074Z ##[debug]   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
2017-11-01T20:42:52.7193075Z ##[debug]   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
2017-11-01T20:42:52.7213056Z ##[debug]   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2017-11-01T20:42:52.7223082Z ##[debug]   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
2017-11-01T20:42:52.7233118Z ##[debug]   at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
2017-11-01T20:42:52.7243066Z ##[debug]   at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
2017-11-01T20:42:52.7253334Z ##[debug]   at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
2017-11-01T20:42:52.7263325Z ##[debug]   at System.Management.Automation.PSScriptCmdlet.DoEndProcessing()
2017-11-01T20:42:52.7273313Z ##[debug]   at System.Management.Automation.CommandProcessorBase.Complete()
2017-11-01T20:42:52.7333107Z ##[error]The term 'Get-ServiceFabricComposeApplicationStatus' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2017-11-01T20:42:52.7333107Z ##[debug]Processed: ##vso[task.logissue type=error]The term 'Get-ServiceFabricComposeApplicationStatus' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
2017-11-01T20:42:52.7353062Z ##[debug]Processed: ##vso[task.complete result=Failed]
2017-11-01T20:42:52.7973056Z ##[section]Finishing: Deploy docker-compose application to a Service Fabric cluster
2017-11-01T20:42:52.8023051Z ##[section]Finishing: Release

**Comment by brandmo** [comment]
The key part of your debug dump is that v0.1.7 of the Service Fabric Compose task is still being used. I think we'll need to wait to v0.2.1 to be deployed before it'll start working again.

**Comment by NCarlsonMSFT** [comment]
We are working on getting version 0.2.0 deployed, in the mean-time if you need to unblock yourself you can enlist is this repo, build the task locally ([steps](https://github.com/Microsoft/vsts-tasks/blob/master/docs/contribute.md)), and deploy it to your VSTS instance uing the [TFS-CLI](https://github.com/Microsoft/tfs-cli#setup)

**Comment by mdegroot71** [comment]
What do we do if we don't have our own VSTS instance?  What's the ETA on getting 0.2.0 deployed?

**Comment by mdegroot71** [comment]
0.2.0 is deployed.  Awesome!  Thanks guys!

**Comment by brandmo** [comment]
Works great thanks :). Now we just need it to support the new update commands, I'll open a new ticket for that and link it here

**Comment by brandmo** [comment]
https://github.com/Azure/service-fabric-issues/issues/598

**Comment by bishal-pdMSFT** [comment]
@NCarlsonMSFT closing this issue.
